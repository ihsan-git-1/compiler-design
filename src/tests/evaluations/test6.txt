class test {
    int x=2;
    int y=3;

    int z = 2+x*y;
    int w = x/2*6;
    int t = y*9+4*x-5;
    int r = y*y+x+x-1*2/4+4;

  }