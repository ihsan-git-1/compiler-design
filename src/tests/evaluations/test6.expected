
*********** AST ************

Line : (1) Top Tree Declaration  Child Count =  1 Type = Top Tree
Line : (1) Class Declaration Name is test parent AllClassesDeclaration Child Count =  10 Type = class
Line : (2) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (2) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 Line : (3) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (3) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 Line : (5) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (5) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 Line : (6) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (6) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 Line : (7) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (7) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 Line : (8) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (8) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer) ,  

*********** SYMBOL TABLE ************

Global Scope (0)
Class Scope (1) Parent Is Global Scope (0)

Identifier x, Type int, Value : 2 , Scope Class Scope (1)
Identifier y, Type int, Value : 3 , Scope Class Scope (1)
Identifier z, Type int, Value : 8 , Scope Class Scope (1)
Identifier w, Type int, Value : 6 , Scope Class Scope (1)
Identifier t, Type int, Value : 30 , Scope Class Scope (1)
Identifier r, Type int, Value : 17 , Scope Class Scope (1)
